app/models/enterprise.rb

```diff
@@ -43,7 +43,9 @@ class Enterprise < ApplicationRecord
                                foreign_key: 'supplier_id',
                                dependent: :destroy
   has_many :supplied_variants, through: :supplied_products, source: :variants
-  has_many :distributed_orders, class_name: 'Spree::Order', foreign_key: 'distributor_id'
+  has_many :distributed_orders, class_name: 'Spree::Order',
+                                foreign_key: 'distributor_id',
+                                dependent: :destroy
```

Comment:
Currently we can't delete enterprises with orders. Supporting that would need a bit more thought and design. So for now, this should be `:raise`. We don't know which other code would break if we have orphaned orders.

Reply:
Noted. I have chosen to go with `dependent: :restrict_with_exception` as it raises an `ActiveRecord::DeleteRestrictionError`.